File: 3619-adjacent-increasing-subarrays-detection-ii/3619-adjacent-increasing-subarrays-detection-ii.cpp
class Solution {
public:
    int maxIncreasingSubarrays(vector<int>& nums) {
        int res = 0;
        int count = 1;
        int preCount = 0;
        for(int i = 1; i < nums.size(); i++)
        {
            if(nums[i] > nums[i - 1])
            {
                count++;
            }
            else
            {
                preCount = count;
                count = 1;
            }
            res = max(res, min(preCount, count));
            res = max(res, count / 2);
        }
        return res;
    }
};

File: 3619-adjacent-increasing-subarrays-detection-ii/3619-adjacent-increasing-subarrays-detection-ii.py
class Solution:
    def maxIncreasingSubarrays(self, nums: List[int]) -> int:
        res = 0
        count = 1
        preCount = 0
        for i in range(1, len(nums)):
            if nums[i] > nums[i - 1]:
                count += 1
            else:
                preCount = count
                count = 1
            res = max(res, min(preCount, count))
            res = max(res, count // 2)
        return res